### PR TITLE
CODENVY-666: Adapt permissions to '*' user wildcard

### PR DESCRIPTION
**Pull request by akorneta**
### What does this PR do?
Adapts permissions components to store '*' as user identifier.
### What issues does this PR fix or reference?
#666 
### Previous Behavior
Permission was stored without '*' as user id

@skabashnyuk, @evoevodin please take a look

### CODE REVIEW

File: assembly/onpremises-ide-packaging-war-platform-api/src/main/resources/META-INF/persistence.xml [hunk 1]
```diff
@@ -58,7 +58,9 @@
         <class>com.codenvy.api.permission.server.model.impl.AbstractPermissions</class>
         <class>com.codenvy.api.workspace.server.model.impl.WorkerImpl</class>
         <class>com.codenvy.api.workspace.server.stack.StackPermissionsImpl</class>
-        <class>com.codenvy.api.workspace.server.recipe.RecipePermissionsImpl</class>
+
+        <class>com.codenvy.api.machine.server.recipe.RecipePermissionsImpl</class>
+
```
Comment (skabashnyuk) [inline assembly/onpremises-ide-packaging-war-platform-api/src/main/resources/META-INF/persistence.xml]: remove extra spaces

File: wsmaster/codenvy-hosted-api-machine/src/main/java/com/codenvy/api/machine/server/filters/RecipePermissionsFilter.java [hunk 2]
```diff
@@ -56,7 +51,7 @@ public void filter(GenericResourceMethod genericResourceMethod, Object[] argumen
             case "getRecipe":
             case "getRecipeScript":
                 recipeId = ((String)arguments[0]);
-                action = READ;
+                action = RecipeDomain.READ;
```
Comment (skabashnyuk) [inline wsmaster/codenvy-hosted-api-machine/src/main/java/com/codenvy/api/machine/server/filters/RecipePermissionsFilter.java]: static import?

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/SetPublicPermissionsFilter.java [hunk 3]
```diff
@@ -14,8 +14,10 @@
  */
 package com.codenvy.api.workspace.server.filters;
 
+//import com.codenvy.api.machine.server.recipe.RecipeDomain;
```
Comment (skabashnyuk) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/filters/SetPublicPermissionsFilter.java]: remove

**Comment by skabashnyuk** [comment]
ok. @evoevodin @mshaposhnik @sleshchenko ?

File: wsmaster/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/model/impl/AbstractPermissions.java [hunk 4]
```diff
@@ -91,6 +98,24 @@ public String getUserId() {
         return actions;
     }
 
+    @PrePersist
```
Comment (evoevodin) [inline wsmaster/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/model/impl/AbstractPermissions.java]: `@PreUpdate` ?

**Comment by evoevodin** [comment]
ok

File: wsmaster/codenvy-hosted-api-machine/src/test/java/com/codenvy/api/machine/server/filters/RecipePermissionsFilterTest.java [hunk 5]
```diff
@@ -104,7 +105,7 @@ public void shouldCheckPermissionsOnStackReading() throws Exception {
 
         assertEquals(response.getStatusCode(), 204);
         verify(service).getRecipe("recipe123");
-        verify(subject).hasPermission(eq("recipe"), eq("recipe123"), eq(READ));
+        verify(subject).hasPermission(eq("recipe"), eq("recipe123"), Matchers.eq(READ));
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-machine/src/test/java/com/codenvy/api/machine/server/filters/RecipePermissionsFilterTest.java]: `Matchers.eq(READ)` -> `eq(READ)` ?

File: wsmaster/codenvy-hosted-api-machine/src/test/java/com/codenvy/api/machine/server/spi/tck/RecipePermissionsDaoTest.java [hunk 6]
```diff
@@ -224,6 +224,13 @@ public void shouldThrowExceptionWhenPermissionsExistsActionArgumentIsNull() thro
         dao.exists("user", "instance", null);
     }
 
+    @Test
+    public void shouldStoreRecipePermission() throws Exception {
+        final RecipePermissionsImpl recipePermissions = new RecipePermissionsImpl("*", "recipe1", Arrays.asList("read", "use", "run"));
+        dao.store(recipePermissions);
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-machine/src/test/java/com/codenvy/api/machine/server/spi/tck/RecipePermissionsDaoTest.java]: Should you check that permissions was stored or just that no exception during permissions storing?

File: wsmaster/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/PermissionsManager.java [hunk 7]
```diff
@@ -80,27 +82,34 @@ public void storePermission(Permissions permissions) throws ServerException, Con
...
+        store(permissionsDao, userId, instanceId, permissions);
+    }
 
-        final AbstractPermissionsDomain permissionsDomain = getDomain(permissions.getDomainId());
-        final AbstractPermissions permissionsEntity = permissionsDomain.newInstance(userId, instanceId, permissions.getActions());
+    private <T extends AbstractPermissions> void store(PermissionsDao<T> dao,
+                                                       String userId,
+                                                       String instanceId,
+                                                       Permissions permissions) throws ConflictException,
+                                                                                         ServerException {
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/PermissionsManager.java]: check formatting here

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/listener/RemoveStackOnLastUserRemovedEventSubscriber.java [hunk 8]
```diff
@@ -14,7 +14,9 @@
  */
 package com.codenvy.api.workspace.server.jpa.listener;
 
-import com.codenvy.api.workspace.server.jpa.JpaStackPermissionsDao;
+
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/jpa/listener/RemoveStackOnLastUserRemovedEventSubscriber.java]: pls remove extra line

File: wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/stack/OnPremisesStackLoader.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+            } catch (Exception e) {
+                LOG.error("Failed to store stacks ", e);
+            }
+        }
+    }
+
+    private void loadStack(StackImpl stack) {
+        setIconData(stack, stackIconFolderPath);
+        try {
+            stackDao.update(stack);
```
Comment (sleshchenko) [inline wsmaster/codenvy-hosted-api-workspace/src/main/java/com/codenvy/api/workspace/server/stack/OnPremisesStackLoader.java]: should we reset permissions here?

**Comment by sleshchenko** [comment]
LGTM

**Pull request merged**